File: examples/python/dependency-inversion-principle/bad.py
class MySQLConnection:
    def connect(self):
        return 'Database connection'

class PasswordReminder:
    def __init__(self, dbConnection: MySQLConnection):
        self.dbConnection = dbConnection


'''
Here is an example of a PasswordReminder that connects to a MySQL database:

'''
conn = MySQLConnection()

passReminder = PasswordReminder(conn)

print(conn.connect())

File: examples/csharp/DependencyInversionPrinciple/Bad.cs
using System;

namespace DIP.Bad
{
    public class MySQLConnection
    {
        public MySQLConnection() { }

        public string Connect()
        {
            return "Database connection";
        }
    }

    public class PasswordReminder
    {
        private MySQLConnection _connection;
        public PasswordReminder(MySQLConnection dbConnection)
        {
            this._connection = dbConnection;
        }
    }
}
